### PR TITLE
Add example for Google Datastore

### PR DESCRIPTION
**Pull request by sebelga**
This is an example to use DataLoader with the Google Datastore

### CODE REVIEW

File: examples/GoogleDatastore.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+const DataLoader = require('dataloader');
+const Datastore = require('@google-cloud/datastore');
+
+const datastore = new Datastore();
+
+/**
+ * Generate a unique string id for each key
+ */
+const toString = toString(key) => {
+    let id = = key.namespace || '';
```
Comment (leebyron) [inline examples/GoogleDatastore.md]: double equals

File: examples/GoogleDatastore.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+```js
+const DataLoader = require('dataloader');
+const Datastore = require('@google-cloud/datastore');
+
+const datastore = new Datastore();
+
+/**
+ * Generate a unique string id for each key
+ */
+const toString = toString(key) => {
```
Comment (leebyron) [inline examples/GoogleDatastore.md]: Perhaps: `function keyToString(key) {`?

Comment (sebelga) [reply examples/GoogleDatastore.md]: Ok, making the change.. 👍

**Comment by leebyron** [comment]
I think the example code you supplied was more complicated than necessary and possibly also incorrect, though I'm not as familiar with GC datastore.

I've made some edits for you to consider. Please take a look at this change and let me know if it looks correct

**Comment by sebelga** [comment]
It is indeed much simpler 👍  I don't know why but for some reason having big keys like `JSON.stringify(key)` in my objects is something I don't think of and I don't normally do. I should change that :)

**Comment by leebyron** [comment]
I fell back on JSON.stringify because I took a look at the documentation for Google Datastore keys, and they apparently can vary considerably and be recursive. I think if you know for sure that you're only using a subset of possible keys, and optimized function can be a bit faster though for a generic example implementation we should assume support for all keys.

**Comment by sebelga** [comment]
Just for info, the "path" I was concatenating is always the **full path** of a key. So effectively it can be of any size but it will always represent the full key (grand parents, parents, child, ..) and so is unique to each key.
I then added the "namespace" which obviously could change and differentiate 2 keys of same path.

**Pull request merged**